tests/phpunit/tests/html/wpHtmlTagProcessor.php

```diff
@@ -0,0 +1,2021 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Tag_Processor functionality.
+ *
+ * @package WordPress
+ * @subpackage HTML
+ */
+
+/**
+ * @group html
+ *
+ * @coversDefaultClass WP_HTML_Tag_Processor
+ */
+class Tests_HTML_wpHtmlTagProcessor extends WP_UnitTestCase {
+	const HTML_SIMPLE       = '<div id="first"><span id="second">Text</span></div>';
+	const HTML_WITH_CLASSES = '<div class="main with-border" id="first"><span class="not-main bold with-border" id="second">Text</span></div>';
+	const HTML_MALFORMED    = '<div><span class="d-md-none" Notifications</span><span class="d-none d-md-inline">Back to notifications</span></div>';
+
+	/**
+	 * @ticket 56299
+	 *
+	 * @covers WP_HTML_Tag_Processor::get_tag
```

Comment:
In talking with @costdev and others, the approach for now is to document `@covers Class_Name::method_name` at the test method. Why?

* Split methods into separate files.
There are plans/discussions to rearchitect the test suites to break up class tests into separate method tests, ie one method in a test class. Using the `@covers Class_Name::method_name` approach can help with that effort to quickly identify how to break out tests. Also once this happens, then annotation at the test method level is no longer needed.

* Confusion. Added code review effort/time. 
I'll use me as an example. When I read a test method and see `@covers ::name` I read it as `@covers ::function_name` without realizing it can also refer to a method name. So I have to scroll to the top of the file to find if there's a `@coversDefaultClass` annotation or not.

* Core uses both approaches.
Currently, Core uses both approaches to annotating `@covers`. There's not one standard or consistent approach. Yet.

A Trac ticket will be opened for broader discussion to collective consensus on the `@covers` coding standard.

Reply:
What do you think @anton-vlasenko @costdev? Is that a good approach for now?

Reply:
@hellofromtonya Absolutely.

@dmsnell Apologies that this means revising the `@covers` annotations in this PR after reviews suggested other changes to them. The approach on which Tonya, myself and others agreed, is a way we can ensure consistency while the community seeks consensus.

Reply:
@hellofromtonya 💯 I'm fine with that approach. 

> A Trac ticket will be opened for broader discussion to collective consensus on the @covers coding standard.

That's great.
I'm all for having a consensus on the `@covers` coding standard.
The sooner it's defined, the fewer tests will have to be refactored.

Reply:
I believe the tag processor test file is done. The bookmark test file will need to changes.